share/spice/forecast/forecast.js

```diff
@@ -158,10 +151,9 @@
         dailyObj[i] = days[i];
         day = days[i];
 
-        tmp_date = new Date();
-        tmp_date.setDate(date_i+i);
-        dailyObj[i].date = tmp_date.toDateString().substr(4,6);
-        dailyObj[i].day = i == 0 ? 'Today' : day_strs[(today_i+i)%7];
+        tmp_date = moment(today);
+        tmp_date.add(i, 'days');
+        dailyObj[i].date = tmp_date.format("MMM D");        
```

Comment:
@jagtalon Agreed, your example reads much better. I've simplified the code more in the latest commit but there are a couple more things to consider:

* The `tmp_date` variable is required to avoid duplicating code at lines [152](https://github.com/duckduckgo/zeroclickinfo-spice/pull/2432/files#diff-f43444f36ddec5dfc14010d40ef963e1R152), [164](https://github.com/duckduckgo/zeroclickinfo-spice/pull/2432/files#diff-f43444f36ddec5dfc14010d40ef963e1R164) & [166](https://github.com/duckduckgo/zeroclickinfo-spice/pull/2432/files#diff-f43444f36ddec5dfc14010d40ef963e1R166).
* I've continued using the `moment(today)` [clone](http://momentjs.com/docs/#/parsing/moment-clone/) to avoid two issues:
  * By calculating todays date [once](https://github.com/duckduckgo/zeroclickinfo-spice/pull/2432/files#diff-f43444f36ddec5dfc14010d40ef963e1R123), we avoid a potential edge case where if this `for` loop is in the middle of running at the stroke of midnight we'd skip a day
  * Adding time to a moment [mutates](http://momentjs.com/docs/#/manipulating/add/) the original moment, hence why `today.add(i, 'days')` couldn't be used at line [151](https://github.com/duckduckgo/zeroclickinfo-spice/pull/2432/files#diff-f43444f36ddec5dfc14010d40ef963e1R151)

Thoughts?